### PR TITLE
Adds useful kdocs for all members declared in AnyElement and other kdocs improvements

### PR DESCRIPTION
**Pull request by popematt**
**Issue #, if available:**

#67

**Description of changes:**

This adds the documentation changes mentioned in #67

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by cheqianh — APPROVED** [review]

File: src/com/amazon/ionelement/api/AnyElement.kt [hunk 1]
```diff
@@ -149,207 +149,411 @@ import java.math.BigInteger
...
+    /**
+     * Attempts to narrow this element to a [IntElement].
+     * If this element is not an Ion `int`, or if it is `null.int`, throws an [IonElementConstraintException].
+     */
     public fun asInt(): IntElement
 
-    /** See [AnyElement]. */
+    /**
+     * Attempts to narrow this element to a [IntElement] or `null`.
+     * If this element is not an Ion `bool` or `null.null`, throws an [IonElementConstraintException].
```
Comment (rmarrowstone) [inline src/com/amazon/ionelement/api/AnyElement.kt:172]: s/bool/int

File: src/com/amazon/ionelement/api/AnyElement.kt [hunk 2]
```diff
@@ -149,207 +149,411 @@ import java.math.BigInteger
...
     public fun asStructOrNull(): StructElement?
 
     /**
      * If this is an Ion integer, returns its [IntElementSize] otherwise, throws [IonElementConstraintException].
      */
     public val integerSize: IntElementSize
 
-    /** See [AnyElement]. */
+    /**
+     * Gets the value of the element, assuming that it is a non-null Ion `bool`.
```
Comment (rmarrowstone) [inline src/com/amazon/ionelement/api/AnyElement.kt:354]: Nit: I wouldn't use the term 'gets' for properties.

**Review by rmarrowstone — APPROVED** [review]

File: src/com/amazon/ionelement/api/AnyElement.kt (hunk 2 above)
Comment (popematt) [reply src/com/amazon/ionelement/api/AnyElement.kt:354]: I was not thrilled with "gets", but I couldn't think of a better way to phrase it. Got any suggestions?

File: src/com/amazon/ionelement/api/AnyElement.kt (hunk 1 above)
Comment (popematt) [reply src/com/amazon/ionelement/api/AnyElement.kt:172]: Good catch—thank you!

File: src/com/amazon/ionelement/api/AnyElement.kt (hunk 2 above)
Comment (popematt) [reply src/com/amazon/ionelement/api/AnyElement.kt:354]: What about something like this?
```suggestion
     * The value of the element, assuming that the element is a non-null Ion `bool`.
     * Attempting to access this property on an element that violates the type assumption will result in an  [IonElementConstraintException].
```

Comment (rmarrowstone) [reply src/com/amazon/ionelement/api/AnyElement.kt:354]: Yea... I want to suggest "Access" though that may be a bit wordy.

**Review by rmarrowstone — APPROVED** [review]
